script.rb

```diff
@@ -0,0 +1,155 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    temp = []
+    i = 0
+    while i < length
+      temp << self[i] if yield(self[i])
+      i += 1
+    end
+    temp
+  end
+
+  def my_all?
+    i = 0
+    while i < length
+      return true if yield(self[i]) && yield(self[i + 1])
+
+      i += 1
+      return false
+    end
+  end
+
+  def my_any?
+    i = 0
+    while i < length
+      return true if yield(self[i]) || yield(self[i + 1])
+
+      i += 1
+      return false
+    end
+  end
+
+  def my_none?
+    i = 0
+    while i < length
+      return true unless yield(self[i])
+
+      i += 1
+      return false
+    end
+  end
+
+  def my_count
+    cont = 0
+    i = 0
+    while i < length
+      cont += 1 if yield(self[i])
+      i += 1
+    end
+    cont
+  end
```

Comment:
- For `my_count`,handle the following cases
1. it should return the number of items in enum through enumeration
 - The built-in method 
`[1,2,3,4].count` returns `4`
 - Your method gives the following error

> LocalJumpError:
        no block given (yield)

2. when an argument is given, it should count the number of items in enum that are equal to the argument provided.

 - The built-in method 
`[3,3,3,3].count(1)` returns `0`

 - Your method gives the following error

> ArgumentError:
        wrong number of arguments (given 1, expected 0)

Reply:
my_count method updated